code/game/gamemodes/cm_initialize.dm

```diff
@@ -1022,8 +1022,11 @@ Additional game mode variables.
 		return
 
 	var/deathtime = world.time - joe_candidate.timeofdeath
-	if((deathtime < JOE_JOIN_DEAD_TIME && (joe_candidate.ckey in joes)) && !MODE_HAS_TOGGLEABLE_FLAG(MODE_BYPASS_JOE))
-		to_chat(joe_candidate, SPAN_WARNING("You have been dead for [DisplayTimeText(deathtime)]. You need to wait <b>[DisplayTimeText(JOE_JOIN_DEAD_TIME - deathtime)]</b> before rejoining as a Working Joe!"))
+	if(((deathtime < JOE_JOIN_DEAD_TIME || MODE_HAS_TOGGLEABLE_FLAG(MODE_DISABLE_JOE_RESPAWN)) && (joe_candidate.ckey in joes)) && !MODE_HAS_TOGGLEABLE_FLAG(MODE_BYPASS_JOE))
```

Comment:
I really do not like this being two flags still. For example here you have MODE_BYPASS_JOE overriding whatever MODE_DISABLE_JOE_RESPAWN is set to. But this is the pitfall of gigantic if statements and why our guidelines push for early returns.

Clean this up and remove the overriding by making the logic for MODE_HAS_TOGGLEABLE_FLAG(MODE_DISABLE_JOE_RESPAWN) as its own if statement.

If MODE_DISABLE_JOE_RESPAWN is set then people who have been a joe cannot respawn as a joe.

If MODE_BYPASS_JOE is set then people who have been a joe don't need to wait for JOE_JOIN_DEAD_TIME.